### PR TITLE
feat(organization): add support for group projects

### PR DESCRIPTION
**Pull request by ivan-savciuc**
## About this change—what it does

Adding support for organization group project permissions

### CODE REVIEW

File: internal/plugin/service/organization/organization_group_project_test.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	"github.com/hashicorp/terraform-plugin-testing/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+
+	acc "github.com/aiven/terraform-provider-aiven/internal/acctest"
+)
+
+func TestAccOrganizationGroupProject(t *testing.T) {
+	orgID := os.Getenv("AIVEN_ORG_ID")
+
+	if orgID == "" {
```
Comment (Serpentiel) [inline internal/plugin/service/organization/organization_group_project_test.go:17]: please replace with `os.LookupEnv`

File: internal/plugin/service/organization/organization_group_project_test.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+		t.Skip("Skipping test due to missing AIVEN_ORG_ID environment variable")
+	}
+
+	if os.Getenv("PROVIDER_AIVEN_ENABLE_BETA") == "" {
```
Comment (Serpentiel) [inline internal/plugin/service/organization/organization_group_project_test.go:21]: same as above

**Review by Serpentiel — APPROVED** [review]
lgtm but please check the review below

**Review by Serpentiel — CHANGES_REQUESTED** [review]
please address the above review comments

File: go.mod [hunk 3]
```diff
@@ -3,7 +3,7 @@ module github.com/aiven/terraform-provider-aiven
 go 1.21
 
 require (
-	github.com/aiven/aiven-go-client/v2 v2.5.0
+	github.com/aiven/aiven-go-client/v2 v2.5.1-0.20231228170105-c4df0fe534c0
```
Comment (Serpentiel) [inline go.mod:6]: please use `v2.6.0`; setting this to a fixed commit will break dependabot's automatic updates

**Review by Serpentiel — DISMISSED** [review]

File: internal/plugin/service/organization/organization_group_project_test.go (hunk 2 above)
Comment (ivan-savciuc) [reply internal/plugin/service/organization/organization_group_project_test.go]: updated with the os.LookupEnv

**Review by Serpentiel — APPROVED** [review]
lgtm

**Pull request merged**